tests/reference/defer_insertion.html

```diff
@@ -0,0 +1,21 @@
+<html>
+  <head>
+    <title>defer_javascript example</title>
+  </head>
+  <body>
+    <script type="application/javascript">
+      var output = [];
+      var headElem = document.getElementsByTagName("head")[0];
+      elt = document.createElement("script");
+      elt.setAttribute("language", "javascript");
+      elt.setAttribute("type", "text/javascript");
+      elt.setAttribute("src", '../assets/js_push1.js');
+      elt.onload = function() {
+        output.push(elt.getAttribute("src"));
```

Comment:
Can't we add some assertions in all the deferjs tests?
I think this will document the expected behavior.

Here, the test assert the behavior is the same with and without the deferjs but I can't tell if you are testing the right thing. According to the issue description, the `src` value should be `../assets/js_push1.js` but perhaps the `push` is misplaced and you are not really testing the fix you have added.

Reply:
Yes, I see what you would like. I didn't insert assertions in the html code because it's tricky to ensure that all assertions will be executed.

I found that it was easier to add element in an variable and assert the content of the variable outside the browser.

I think that the tests stay minimal. In this case, js_push1.js should be executed and the src attribute should have the updated value (i.e js_push2.js).

Reply:
I don't know the test framework we use here but isn't there any way to trigger an event from the HTML page when all conditions are met and let the test framework catch this event.
If the event is not triggered, the test should timeout.